operators/src/processing/interpolation/mod.rs

```diff
@@ -0,0 +1,824 @@
+        start_time: TimeInstance,
+    ) -> Result<Option<RasterQueryRectangle>> {
+        // enlarge the spatial bounds in order to have the neighbor pixels for the interpolation
+        let spatial_bounds = tile_info.spatial_partition();
+        let enlarge: Coordinate2D = (self.input_resolution.x, -self.input_resolution.y).into();
+        let spatial_bounds = SpatialPartition2D::new(
+            spatial_bounds.upper_left(),
+            spatial_bounds.lower_right() + enlarge,
+        )?;
+
+        Ok(Some(RasterQueryRectangle {
+            spatial_bounds,
+            time_interval: TimeInterval::new_instant(start_time)?,
+            spatial_resolution: self.input_resolution,
+        }))
+    }
+
+    fn fold_method(&self) -> Self::FoldMethod {
+        self.fold_fn.clone()
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct InterpolationAccu<T: Pixel, I: InterpolationAlgorithm<T>> {
+    pub output_tile: RasterTile2D<T>,
+    pub input_tile: RasterTile2D<T>,
+    pub pool: Arc<ThreadPool>,
+    phantom: PhantomData<I>,
+}
+
+impl<T: Pixel, I: InterpolationAlgorithm<T>> InterpolationAccu<T, I> {
+    pub fn new(
+        input_tile: RasterTile2D<T>,
+        output_tile: RasterTile2D<T>,
+        pool: Arc<ThreadPool>,
+    ) -> Self {
+        InterpolationAccu {
+            input_tile,
+            output_tile,
+            pool,
+            phantom: Default::default(),
+        }
+    }
+}
+
+#[async_trait]
+impl<T: Pixel, I: InterpolationAlgorithm<T>> FoldTileAccu for InterpolationAccu<T, I> {
+    type RasterType = T;
+
+    async fn into_tile(self) -> Result<RasterTile2D<Self::RasterType>> {
+        // now that we collected all the input tile pixels we perform the actual interpolation
+        let mut output_tile = self.output_tile.into_materialized_tile();
+
+        let output_tile = crate::util::spawn_blocking_with_thread_pool(self.pool, move || {
+            I::interpolate(&self.input_tile, &mut output_tile).map(|_| output_tile)
+        })
+        .await??;
+
+        Ok(output_tile.into())
+    }
+
+    fn thread_pool(&self) -> &Arc<ThreadPool> {
+        &self.pool
+    }
+}
+
+impl<T: Pixel, I: InterpolationAlgorithm<T>> FoldTileAccuMut for InterpolationAccu<T, I> {
+    fn tile_mut(&mut self) -> &mut RasterTile2D<T> {
+        &mut self.input_tile
+    }
+}
+
+pub fn create_accu<T: Pixel, I: InterpolationAlgorithm<T>>(
+    tile_info: TileInformation,
+    query_rect: RasterQueryRectangle,
+    no_data_value: T,
+    pool: Arc<ThreadPool>,
+    tiling_specification: TilingSpecification,
+) -> impl Future<Output = Result<InterpolationAccu<T, I>>> {
+    // create an accumulator as a single tile that fits all the input tiles
+    crate::util::spawn_blocking(move || {
+        let tiling = tiling_specification.strategy(
+            query_rect.spatial_resolution.x,
+            -query_rect.spatial_resolution.y,
+        );
+
+        // TODO: enlarge query_rect to be sure to have all the neighbor pixels for the edges
```

Comment:
is that still needed?

Reply:
I removed it 9fba3541e7ffdc6b094d5a270c28ddcc36909909